source/rules/no-subclass.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows subclassing RxJS classes.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Subclassing RxJS classes is forbidden."
+    },
+    schema: []
+  },
+  create: context => {
+    const { couldBeType } = typecheck(context);
+
+    const queryNames = [
+      "AsyncSubject",
+      "BehaviorSubject",
+      "Observable",
+      "ReplaySubject",
+      "Scheduler",
+      "Subject",
+      "Subscriber"
+    ];
+
+    return {
+      [`ClassDeclaration[superClass.name=/^(${queryNames.join(
+        "|"
+      )})$/] > Identifier.superClass`]: (node: es.Identifier) => {
+        if (
+          queryNames.some(name =>
+            couldBeType(node, name, { name: /[\/\\]rxjs[\/\\]/ })
```

Comment:
This is different from TSLint, tere we have to check on every type.  
In TSLint `Subject` is detected with `couldBeObservable`, in ESLint this is not the case.
Maybe something we have to look into?

Also, Why are we checking on `rxjs` here and not in other tests?

Reply:
> Why are we checking on rxjs here and not in other tests?

Because of this issue: https://github.com/cartant/rxjs-tslint-rules/issues/87

In this rule, the type name is all there is to go on. IIRC, in other rules, whether or not something is an observable is not the sole point of the rule and there are other things that have to 'match'. Also, with `/rxjs/` checked in this rule, the rule won't catch the subclassing of re-exported exports. That's just the way it is, but we might not want that behaviour elsewhere - it could make other rules less enforceable.

We can have a closer look at this later. I'll create a reminder issue.

Reply:
> In TSLint Subject is detected with couldBeObservable, in ESLint this is not the case.
Maybe something we have to look into?

You've included all of the subject classes in `queryNames`, so what's in this rule LGTM.

Reply:
Thanks for the info